editor/editor_settings.cpp

```diff
@@ -614,13 +614,23 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 	_initial_set("text_editor/completion/use_single_quotes", false);
 	_initial_set("text_editor/completion/colorize_suggestions", true);
 
+	// External
+	_initial_set("text_editor/external/use_external_editor", false);
+	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_GLOBAL_FILE, "text_editor/external/exec_path", "", "");
+	_initial_set("text_editor/external/exec_flags", "{file}");
+
 	// Help
 	_initial_set("text_editor/help/show_help_index", true);
 	EDITOR_SETTING(Variant::INT, PROPERTY_HINT_RANGE, "text_editor/help/help_font_size", 16, "8,48,1")
 	EDITOR_SETTING(Variant::INT, PROPERTY_HINT_RANGE, "text_editor/help/help_source_font_size", 15, "8,48,1")
 	EDITOR_SETTING(Variant::INT, PROPERTY_HINT_RANGE, "text_editor/help/help_title_font_size", 23, "8,64,1")
 	EDITOR_SETTING(Variant::INT, PROPERTY_HINT_ENUM, "text_editor/help/class_reference_examples", 0, "GDScript,C#,GDScript and C#")
 
+	/* Dotnet*/
+
+	// Editor
+	EDITOR_SETTING(Variant::INT, PROPERTY_HINT_ENUM, "dotnet/editor/external_editor", 0, "Disabled,Visual Studio,MonoDevelop,Visual Studio Code,JetBrains Rider,Custom");
```

Comment:
This is defined in `GodotSharpEditor`:

https://github.com/godotengine/godot/blob/066e7d483a8f699fd94e7133ac23978cd460a72d/modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs#L571-L577

And the hint string depends on the platform (for example Visual Studio is not available in Linux so it's not in the hint).